### PR TITLE
util: Hint `ApiMetadata.message_class` with `Optional[Type[Streamable]]`

### PR DESCRIPTION
**Pull request by xdustinface**
Im working towards more strict mypy in `chia.server.ws_connection` and  this change helps there while (i think) it still retains the same  behaviour.

### CODE REVIEW

File: chia/util/api_decorators.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ApiMetadata:
     bytes_required: bool = False
     execute_task: bool = False
     reply_types: List[ProtocolMessageTypes] = field(default_factory=list)
-    message_class: Optional[Any] = None
+    message_class: Any = None
```
Comment (arvidn) [inline chia/util/api_decorators.py:30]: I think `Optional[]` adds value here; it's a more specific type than just `Any`.
A better type would be:
```
message_class: Optional[Type[Streamable]] = None
```

**Review by arvidn — CHANGES_REQUESTED** [review]
I would like to hear from @altendky too

File: chia/util/api_decorators.py (hunk 1 above)
Comment (xdustinface) [reply chia/util/api_decorators.py]: Right, `Type[Streamable]` makes sense.

**Comment by altendky** [comment]
> I would like to hear from @altendky too

I don't remember anything offhand. More specific sounds good if mypy is happy.  If I come up with a problem and this has been merged... Well, we aren't exactly running out of PRs.

**Review by arvidn — APPROVED** [review]

**Pull request merged**